src/composeMocks.ts

```diff
@@ -14,53 +19,119 @@ interface PublicAPI {
   stop(): Promise<void>
 }
 
-const createStart = (
-  worker: ServiceWorker,
-  workerRegistration: ServiceWorkerRegistration,
-): PublicAPI['start'] => {
+const activateMocking = (worker: ServiceWorker) => {
+  worker.postMessage('MOCK_ACTIVATE')
+
+  return new Promise((resolve, reject) => {
+    // Wait until the mocking is enabled to resolve the start Promise
+    addMessageListener(
+      'MOCKING_ENABLED',
+      () => {
+        console.groupCollapsed(
```

Comment:
Moving the successful activation message to the client-side, so the service worker is not subjected to integrity changes whenever the message's content is changed (contains potentially dynamic links to docs).